### DISCUSSION TITLE
Using vector images (SVG files) for texture rendering (or possible alternatives)

### DISCUSSION THREAD

**Discussion by JoeAnonymous** [bnpr/Malt#418] [Feedback & Feature Requests]
Hello,
         Is there any plan to allow people to use SVG or other vector formats to render textures to 3D models? I do not wish to have to manually convert them into raster images.

I can only think of a solution that relies on the object knowing the distance changes from it and the camera, and updating a buffer image that holds a raster texture, but I do not think that would be possible.

**Comment by pragma37** [comment]
That would require auto-generating a shader function that can compute that specific SVG file result given a UV coordinate, and then run that function for each pixel that wants to "read" the texture.

So it's not impossible, but it would be a huge amount of work to implement and the resulting performance would be far from great.

Here's a web that can generate a shadertoy from an SVG file (paths only) if you want to give it a try:
https://zduny.github.io/shadertoy-svg/

Depending on your specific use case there may be better alternatives, but converting them to raster textures is the only "generic" solution.

**Reply by JoeAnonymous** [reply]
I probably should've specified, but I'm referring to UV unwrapped models. not actually converting a vector into a 3d model, but using them like this https://www.youtube.com/watch?v=jXFp3wPLUPs

Sorry for the confusion.

**Reply by pragma37** [reply]
I know, that was what I was answering to.

For converting it into a 3d model there's already a built-in SVG importer in Blender.

**Reply by JoeAnonymous** [reply]
Ok then, thank you.
